File: src/gestorAplicacion/personas/Enfermedad.java
/* Autores: Diego Andres Gracia Granados, Daniel Giraldo Vanegas, Elian David Velandia Riveros, Juan Camilo Gutierrez Martinez
y Santiago Arboleda Acevedo */

package gestorAplicacion.personas;

import java.io.Serializable;
import java.util.ArrayList;

/* Clase que crea enfermedades, las cuales son usadas para recomendarle al paciente
distintos servicios que estén relacionados con su enfermedad */
public class Enfermedad implements Serializable {
    // Atributos
    private String especialidad;
    private String nombre;
    private String tipologia;
    private int enfermos = 1;
    // Lista de las enfermedades registradas en el hospital
    private static ArrayList<Enfermedad> enfermedadesRegistradas = new ArrayList<>();

    // Constructor
    public Enfermedad(String especialidad, String nombre, String tipologia) {
        this.especialidad = especialidad;
        this.nombre = nombre;
        this.tipologia = tipologia;
        enfermedadesRegistradas.add(this);
    }

    // Métodos

    // Método que aumenta los enfermos en una unidad
    public void nuevoEnfermo(){
        enfermos+=1;
    }

    // Getters y Setters
    public String getEspecialidad() {
        return especialidad;
    }

    public void setEspecialidad(String especialidad) {
        this.especialidad = especialidad;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getTipologia() {
        return tipologia;
    }

    public void setTipologia(String tipologia) {
        this.tipologia = tipologia;
    }

    public static ArrayList<Enfermedad> getEnfermedadesRegistradas() {
        return enfermedadesRegistradas;
    }

    public static void setEnfermedadesRegistradas(ArrayList<Enfermedad> enfermedadesRegistradas) {
        Enfermedad.enfermedadesRegistradas = enfermedadesRegistradas;
    }

    public int getEnfermos() {
        return enfermos;
    }

    public void setEnfermos(int enfermos) {
        this.enfermos = enfermos;
    }

    // toString
    @Override
    public String toString(){
        return nombre + " " + tipologia + " Especialidad que la trata: " + especialidad;
    }
}

File: Python/src/gestor_aplicacion/personas/enfermedad.py
# Autores: Diego Andres Gracia Granados, Daniel Giraldo Vanegas, Elian David Velandia Riveros, Juan Camilo Gutierrez Martinez
# y Santiago Arboleda Acevedo

# Clase que crea enfermedades, las cuales son usadas para recomendarle al paciente
# distintos servicios que estén relacionados con su enfermedad
class Enfermedad:

    _enfermedades_registradas = [] # Atributo de clase

    # Atributos y constructor
    def __init__(self, nombre, tipologia, especialidad):
        self._especialidad = especialidad
        self._nombre = nombre
        self._tipologia = tipologia
        self._enfermos = 1
        Enfermedad._enfermedades_registradas.append(self)

    # Metodos

    # Aumenta en el atributo enfermos en una unidad
    def nuevo_enfermo(self):
        self._enfermos += 1

    def __str__(self):
        return f"{self._nombre} {self._tipologia}"

    @property
    def tipologia(self):
        return self._tipologia

    @tipologia.setter
    def tipologia(self, value):
        self._tipologia = value

    @property
    def enfermos(self):
        return self._enfermos

    @enfermos.setter
    def enfermos(self, value):
        self._enfermos = value

    @property
    def nombre(self):
        return self._nombre

    @nombre.setter
    def nombre(self, value):
        self._nombre = value

    @property
    def especialidad(self):
        return self._especialidad

    @especialidad.setter
    def especialidad(self, value):
        self._especialidad = value

    @classmethod
    def getEnfermedadesRegistradas(cls):
        return cls._enfermedades_registradas

